src/lcm/LifeCycleActions.cc

```diff
@@ -1916,3 +2023,174 @@ void LifeCycleManager::trigger_updatesg(int sgid)
+                    {
+                        // Can't live update the NIC, wrong VM state
+                        case VirtualMachine::BOOT:
+                        case VirtualMachine::BOOT_MIGRATE:
+                        case VirtualMachine::BOOT_SUSPENDED:
+                        case VirtualMachine::BOOT_STOPPED:
+                        case VirtualMachine::BOOT_UNDEPLOY:
+                        case VirtualMachine::BOOT_POWEROFF:
+                        case VirtualMachine::BOOT_UNKNOWN:
+                        case VirtualMachine::BOOT_FAILURE:
+                        case VirtualMachine::BOOT_MIGRATE_FAILURE:
+                        case VirtualMachine::BOOT_UNDEPLOY_FAILURE:
+                        case VirtualMachine::BOOT_STOPPED_FAILURE:
+                        case VirtualMachine::MIGRATE:
+                        case VirtualMachine::HOTPLUG_NIC:
+                        case VirtualMachine::HOTPLUG_NIC_POWEROFF:
+                        case VirtualMachine::HOTPLUG:
+                        case VirtualMachine::HOTPLUG_SNAPSHOT:
+                        case VirtualMachine::HOTPLUG_SAVEAS:
+                        case VirtualMachine::HOTPLUG_RESIZE:
+                        case VirtualMachine::DISK_SNAPSHOT:
+                        case VirtualMachine::DISK_SNAPSHOT_DELETE:
+                        case VirtualMachine::DISK_RESIZE:
+                        case VirtualMachine::UNKNOWN:
+                            is_error = true;
+                            break;
+
+                        // Update VM information + virtual network at host
+                        case VirtualMachine::RUNNING:
+                            is_update = true;
+                            break;
+
+                        // Update just VM information
+                        default:
+                            is_tmpl = true;
+                            break;
+                    }
+                }
+
+                // -------------------------------------------------------------
+                // Update VM template with the new NIC attributes & trigger update
+                // -------------------------------------------------------------
+                if ( is_tmpl || is_update )
+                {
+                    rc = vm->nic_update(vnid);
+
+                    if (rc < 0)
+                    {
+                        NebulaLog::error("LCM", "VM " + to_string(vm->get_oid())
+                                         + ": Unable to update Virtual Network " + to_string(vnid));
+
+                        is_error = true;
+                        is_tmpl = is_update = false;
+                    }
+
+                    if (is_update && vm->get_template_attribute("VNET_UPDATE") == nullptr)
+                    {
+                        // Do not trigger driver update in case no attribute was updated
+                        is_update = false;
+                        is_tmpl = true;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We observed that the presence of non-empty statements like `a + b` without clear context or purpose within the codebase can lead to potential confusion or misinterpretation. To mitigate this issue, it&#39;s advisable to ensure that such statements are appropriately integrated into larger constructs that contribute meaningfully to the program&#39;s logic or functionality. This could involve assigning the result of the expression to a variable.